plugins/modules/aci_l2out.py

```diff
@@ -0,0 +1,322 @@
+     type: dict
+     sample:
+       {
+           "fvTenant": {
+               "attributes": {
+                   "descr": "Production environment",
+                   "name": "production"
+               }
+           }
+       }
+   filter_string:
+     description: The filter string used for the request
+     returned: failure or debug
+     type: str
+     sample: ?rsp-prop-include=config-only
+   method:
+     description: The HTTP method used for the request to the APIC
+     returned: failure or debug
+     type: str
+     sample: POST
+   response:
+     description: The HTTP response from the APIC
+     returned: failure or debug
+     type: str
+     sample: OK (30 bytes)
+   status:
+     description: The HTTP status from the APIC
+     returned: failure or debug
+     type: int
+     sample: 200
+   url:
+     description: The HTTP url used for the request to the APIC
+     returned: failure or debug
+     type: str
+     sample: https://10.11.12.13/api/mo/uni/tn-production.json
+   '''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        bd=dict(type='str'),
+        l2out=dict(type='str', aliases=['name']),
+        domain=dict(type='str'),
+        vlan=dict(type='int'),
+        description=dict(type='str'),
+        target_dscp=dict(type='str'),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+        tenant=dict(type='str'),
+        name_alias=dict(type='str'),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['l2out', 'tenant']],
+            ['state', 'present', ['bd', 'l2out', 'tenant', 'domain', 'vlan']],
+        ],
+    )
+
+    bd = module.params['bd']
+    l2out = module.params['l2out']
+    description = module.params['description']
+    domain = module.params['domain']
+    vlan = module.params['vlan']
+    state = module.params['state']
+    tenant = module.params['tenant']
+    target_dscp = module.params['target_dscp']
+    child_classes = ['l2extRsEBd', 'l2extRsL2DomAtt', 'l2extLNodeP']
+    name_alias = module.params.get('name_alias')
+
+    aci = ACIModule(module)
+    aci.construct_url(
+        root_class=dict(
+            aci_class='fvTenant',
+            aci_rn='tn-{0}'.format(tenant),
+            module_object=tenant,
+            target_filter={'name': tenant},
+        ),
+        subclass_1=dict(
+            aci_class='l2extOut',
+            aci_rn='l2out-{0}'.format(l2out),
+            module_object=l2out,
+            target_filter={'name': l2out},
+        ),
+        child_classes=child_classes,
+    )
+
+    aci.get_existing()
+
+    if state == 'present':
+        child_configs = [
+            dict(l2extRsL2DomAtt=dict(attributes=dict(
```

Comment:
we need to fix this indentation. It looks bad.

Reply:
I tried to move the respective dicts to new lines but the sanity test complained about it. Let me check

Reply:
Can you make each in a single line?